src/asset-swapper/quote_consumers/feature_rules/transform_erc20_rule.ts

```diff
@@ -239,4 +188,124 @@ export class TransformERC20Rule extends AbstractFeatureRule {
             gasOverhead,
         };
     }
+
+    private createFillQuoteTransformations(quote: SwapQuote, opts: ExchangeProxyContractOpts): ERC20Transformation[] {
+        const transformations = [...this.createTwoHopTransformations(quote, opts)];
+
+        const nonTwoHopTransformation = this.createNonTwoHopTransformation(quote, opts);
+        if (nonTwoHopTransformation !== undefined) {
+            transformations.push(nonTwoHopTransformation);
+        }
+
+        return transformations;
+    }
+
+    private createTwoHopTransformations(quote: SwapQuote, opts: ExchangeProxyContractOpts): ERC20Transformation[] {
+        // This transformer will fill the quote.
+        // TODO: handle `shouldSellEntireBalance` outside.
+        const { refundReceiver, shouldSellEntireBalance } = opts;
+        const { sellToken, buyToken, maxSlippage } = this.getSwapContext(quote, opts);
+        const slippedTwoHopOrders = quote.path.getSlippedOrdersByType(maxSlippage).twoHopOrders;
+
+        return _.flatMap(slippedTwoHopOrders, ({ firstHopOrder, secondHopOrder }) => {
+            const intermediateToken = firstHopOrder.makerToken;
+            return [
+                {
+                    deploymentNonce: this.transformerNonces.fillQuoteTransformer,
+                    data: encodeFillQuoteTransformerData({
+                        side: FillQuoteTransformerSide.Sell,
+                        sellToken,
+                        buyToken: intermediateToken,
+                        ...getFQTTransformerDataFromOptimizedOrders([firstHopOrder]),
+                        refundReceiver: refundReceiver || NULL_ADDRESS,
+                        fillAmount:
+                            !isBuyQuote(quote) && shouldSellEntireBalance ? MAX_UINT256 : firstHopOrder.takerAmount,
```

Comment:
fyi -- i don't see `!isBuyQuote(quote)` in the original code. not sure if this was intentional.